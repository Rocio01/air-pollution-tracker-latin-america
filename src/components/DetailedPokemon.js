import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router';
import { fetchPokemons } from '../redux/pokemons/pokemons';

const DetailedPokemon = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchPokemons());
  }, []);
  const pokemonsState = useSelector((state) => state.pokemonsReducer.pokemons);
  const { name } = useParams();
  return (

    <div className="details">
      <div className="card">
        {(pokemonsState.filter(((pokemon) => pokemon.name === name)).map((pokemon) => (

          <div className="col-sm-6 country-card" key={pokemon.name}>
            <div className="card">
              <img src={pokemon.sprites.front_shiny} className="card-img-top pokemon" alt="pokemon" />
              <div className="card-body">
                <h4 className="card-title">
                  { `Name: ${pokemon.name}`}
                </h4>
                <p>
                  { `Expierence: ${pokemon.base_experience}`}
                </p>

                <p>
                  { `Ability: ${pokemon.abilities[0].ability.name}`}
                </p>
              </div>
            </div>
          </div>
        )))}
      </div>
    </div>
  );
};
export default DetailedPokemon;
