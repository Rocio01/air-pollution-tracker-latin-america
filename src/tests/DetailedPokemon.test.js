import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import DetailedPokemon from '../components/DetailedPokemon';
import DetailedPokemonMock from './mocks/DetailedPokeMock';
import store from '../redux/configureStore';

describe('Detailed pokemon component', () => {
  test('matches Detailed component snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <DetailedPokemon />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Do not render unfiltered pokemon', () => {
    const component = render(
      <DetailedPokemonMock
        name="Raichu"
      />,
    );
    expect(component.container).not.toHaveTextContent('Pikachu');
    expect(component.container).not.toHaveTextContent('100');
  });

  test('Render the filtered pokemon', () => {
    const component = render(
      <DetailedPokemonMock
        name="Pikachu"
      />,
    );
    expect(component.container).toHaveTextContent('Pikachu');
    expect(component.container).toHaveTextContent('100');
  });
});
