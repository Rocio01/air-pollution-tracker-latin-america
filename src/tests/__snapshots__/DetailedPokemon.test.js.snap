// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detailed pokemon component matches Detailed component snapshot 1`] = `
<div
  className="details"
/>
`;
